database-connector/src/main/java/life/qbic/projectmanagement/experiment/persistence/SamplePreviewJpaRepository.java

```diff
@@ -40,13 +45,133 @@ public List<SamplePreview> queryByExperimentId(ExperimentId experimentId, int of
       }
       return order;
     }).toList();
-    return samplePreviewRepository.findSamplePreviewByExperimentId(experimentId,
+
+    Specification<SamplePreview> filterSpecification = generateExperimentIdandFilterSpecification(
+        experimentId, filter);
+    return samplePreviewRepository.findAll(filterSpecification,
         new OffsetBasedRequest(offset, limit, Sort.by(orders))).getContent();
   }
 
   @Override
-  public int queryCountByExperimentId(ExperimentId experimentId) {
-    return samplePreviewRepository.countSamplePreviewsByExperimentId(experimentId);
+  public int queryCountByExperimentId(ExperimentId experimentId, String filter) {
+    Specification<SamplePreview> filterSpecification = generateExperimentIdandFilterSpecification(
+        experimentId, filter);
+    return (int) samplePreviewRepository.count(filterSpecification);
+  }
+
+  private Specification<SamplePreview> generateExperimentIdandFilterSpecification(
+      ExperimentId experimentId, String filter) {
+    Specification<SamplePreview> isBlankSpec = SamplePreviewSpecs.isBlank(filter);
+    Specification<SamplePreview> experimentIdSpec = SamplePreviewSpecs.experimentIdEquals(
```

Comment:
is this experiment name or really id? and why is it needed (especially as "equals") for the sample table?

Reply:
This specification is the ID of the experiment since that is the unique identifier. The idea here is to only fetch the samplepreview information for that specific id(which is why it's an equal statement) :)